publishing/SiteObject.py

```diff
@@ -0,0 +1,197 @@
+'''Classes for files published to S3'''
+
+import mimetypes
+import hashlib
+import gzip
+import binascii
+
+from os import path
+from datetime import datetime
+
+
+mimetypes.init()  # must initialize mimetypes
+
+
+def remove_prefix(text, prefix):
+    '''Returns a copy of text with the given prefix removed'''
+    if text.startswith(prefix):
+        return text[len(prefix):]
+    return text
+
+
+class SiteObject():
+    '''
+    An abstract class for an individual object that can be uploaded to S3
+    '''
+
+    def __init__(self, filename, md5, site_prefix='', dir_prefix=''):
+        self.filename = filename
+        self.md5 = md5
+        self.dir_prefix = dir_prefix
+        self.site_prefix = site_prefix
+
+    @property
+    def s3_key(self):
+        '''The object's key in the S3 bucket'''
+        filename = self.filename
+        if self.dir_prefix:
+            filename = remove_prefix(filename,
+                                     path.join(self.dir_prefix, ''))
```

Comment:
what's the purpose of `path.join(self.dir_prefix, '')` vs. just `self.dir_prefix`? just to add a `/` if the dir_prefix doesn't have one already?

Reply:
yup, exactly